### PR TITLE
BUGFIX EXTREST-137 simpleXml fix for & in values

### PR DESCRIPTION
**Pull request by vdiomidov**

### CODE REVIEW

File: src/Api/Client.php [hunk 1]
```diff
@@ -346,7 +346,7 @@ protected function _arrayToXml(array $array, SimpleXMLElement $xml, $parentEl =
             if (is_array($value)) {
                 $this->_arrayToXml($value, $this->_isAssocArray($value) ? $xml->addChild($el) : $xml, $el);
             } else {
-                $xml->addChild($el, $value);
+                $xml->addChild($el, htmlspecialchars($value));
```
Comment (xgin) [inline src/Api/Client.php:349]: why not `$xml->{$el} = $value`?

File: src/Api/Operator/Mail.php [hunk 2]
```diff
@@ -28,7 +28,7 @@ public function create($name, $siteId, $mailbox = false, $password = '')
             $mailname->addChild('mailbox')->addChild('enabled', 'true');
         }
         if (!empty($password)) {
-            $mailname->addChild('password')->addChild('value', $password);
+            $mailname->addChild('password')->addChild('value', htmlspecialchars($password));
```
Comment (xgin) [inline src/Api/Operator/Mail.php:31]: `$mailname->addChild('password')->value = $password`

File: src/Api/Operator/ProtectedDirectory.php [hunk 3]
```diff
@@ -83,7 +83,7 @@ public function addUser($protectedDirectory, $login, $password)
 
         $info->addChild('pd-id', $protectedDirectory->id);
         $info->addChild('login', $login);
-        $info->addChild('password', $password);
+        $info->addChild('password', htmlspecialchars($password));
```
Comment (xgin) [inline src/Api/Operator/ProtectedDirectory.php:86]:
```
$info->{'pd-id'} = $protectedDirectory->id;
$info->login = $login;
$info->password = $password;
```

File: src/Api/Operator/ProtectedDirectory.php [hunk 4]
```diff
@@ -114,6 +114,7 @@ private function _get($command, $field, $value)
         $filterTag = $getTag->addChild('filter');
         if (!is_null($field)) {
             $filterTag->addChild($field, $value);
```
Comment (xgin) [inline src/Api/Operator/ProtectedDirectory.php:116]: remove this extra element

File: src/Api/Operator/Database.php [hunk 5]
```diff
@@ -136,6 +136,7 @@ private function _get($command, $field, $value)
         $filterTag = $getTag->addChild('filter');
         if (!is_null($field)) {
             $filterTag->addChild($field, $value);
```
Comment (xgin) [inline src/Api/Operator/Database.php:138]: remove this extra element

File: src/Api/Operator/Site.php [hunk 6]
```diff
@@ -33,7 +33,7 @@ public function create(array $properties)
             foreach ($properties[static::PROPERTIES_HOSTING] as $name => $value) {
                 $propertyNode = $hostingNode->addChild('property');
                 $propertyNode->addChild('name', $name);
-                $propertyNode->addChild('value', $value);
+                $propertyNode->addChild('value', htmlspecialchars($value));
```
Comment (xgin) [inline src/Api/Operator/Site.php:36]:
```
$propertyNode->name = $name;
$propertyNode->value = $value;
```

File: src/Api/Operator/Subdomain.php [hunk 7]
```diff
@@ -22,11 +22,11 @@ public function create($properties)
                 foreach ($value as $propertyName => $propertyValue) {
                     $property = $info->addChild($name);
                     $property->addChild('name', $propertyName);
-                    $property->addChild('value', $propertyValue);
```
Comment (xgin) [inline src/Api/Operator/Subdomain.php:25]:
```
$propertyNode->name = $name;
$propertyNode->value = $value;
```

File: src/Api/Operator/Webspace.php [hunk 8]
```diff
@@ -83,7 +83,7 @@ public function create(array $properties, array $hostingProperties = null, strin
             foreach ($hostingProperties as $name => $value) {
                 $property = $infoHosting->addChild('property');
                 $property->addChild('name', $name);
-                $property->addChild('value', $value);
+                $property->addChild('value', htmlspecialchars($value));
```
Comment (xgin) [inline src/Api/Operator/Webspace.php:86]:
```
$property->name = $name;
$property->value = $value;
```

File: tests/Utility/PasswordProvider.php [hunk 9]
```diff
@@ -5,5 +5,5 @@
 
 class PasswordProvider
 {
-    const STRONG_PASSWORD = 'test-PWD*1@42!13#';
+    const STRONG_PASSWORD = 'test-&PWD*1@42!13#';
```
Comment (xgin) [inline tests/Utility/PasswordProvider.php:8]: 👍

File: src/Api/Client.php (hunk 1 above)
Comment (vdiomidov) [reply src/Api/Client.php]: then we get the problem
_It is not yet possible to assign complex types to properties_

Comment (xgin) [reply src/Api/Client.php:349]: hmm, does it work with `htmlspecialchars`? It seems to cast string type for any complex type.

**Pull request merged**